### PR TITLE
Adding an example of routing usage

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this pull request

This pull request (PR) provides an example of a new route can be created and how the custom router works.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.2/extension-dev-guide/routing.html
- https://devdocs.magento.com/guides/v2.3/extension-dev-guide/routing.html

## Links to Magento source code
- New route https://github.com/magento/magento2/blob/2.3/app/code/Magento/Robots/etc/frontend/routes.xml
- Registering a custom router https://github.com/magento/magento2/blob/2.3/app/code/Magento/Robots/etc/frontend/di.xml
- Custom Router https://github.com/magento/magento2/blob/2.3/app/code/Magento/Robots/Controller/Router.php#L56

whatsnew
Added example about how to [declare and load a new route](https://devdocs.magento.com/guides/v2.3/extension-dev-guide/routing.html).

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/extension-dev-guide/routing.md [hunk 1]
```diff
@@ -175,6 +175,172 @@ Name | Description
...
+     * @return ResponseInterface|ResultInterface|Page
+     */
+    public function execute()
+    {
+        return $this->pageFactory->create();
+    }
+}
+```
+
+In the end, let's create the router class, that will match the our custom route name `learning` with the existing `routing` route.
```
Comment (erikmarr) [inline guides/v2.2/extension-dev-guide/routing.md]:
```suggestion
In the end, let's create the router class, that will match the custom route name `learning` with the existing `routing` route.
```

File: guides/v2.2/extension-dev-guide/routing.md [hunk 2]
```diff
@@ -175,6 +175,172 @@ Name | Description
...
+
+            return $this->actionFactory->create(Forward::class, ['request' => $request]);
+        }
+
+        return null;
+    }
+}
+```
+
+As result, by accessing the `http://site.com/learning` route, the `http://site.com/routing/index/index` route is loaded.
```
Comment (erikmarr) [inline guides/v2.2/extension-dev-guide/routing.md]:
```suggestion
As a result, by accessing the `http://site.com/learning` route, the `http://site.com/routing/index/index` route is loaded.
```

File: guides/v2.3/extension-dev-guide/routing.md [hunk 3]
```diff
@@ -179,6 +179,172 @@ Name | Description
...
+     * @return ResponseInterface|ResultInterface|Page
+     */
+    public function execute()
+    {
+        return $this->pageFactory->create();
+    }
+}
+```
+
+In the end, let's create the router class, that will match the our custom route name `learning` with the existing `routing` route.
```
Comment (erikmarr) [inline guides/v2.3/extension-dev-guide/routing.md]:
```suggestion
In the end, let's create the router class, that will match the custom route name `learning` with the existing `routing` route.
```

File: guides/v2.3/extension-dev-guide/routing.md [hunk 4]
```diff
@@ -179,6 +179,172 @@ Name | Description
...
+
+            return $this->actionFactory->create(Forward::class, ['request' => $request]);
+        }
+
+        return null;
+    }
+}
+```
+
+As result, by accessing the `http://site.com/learning` route, the `http://site.com/routing/index/index` route is loaded.
```
Comment (erikmarr) [inline guides/v2.3/extension-dev-guide/routing.md]:
```suggestion
As a result, by accessing the `http://site.com/learning` route, the `http://site.com/routing/index/index` route is loaded.
```

**Comment by eduard13** [comment]
Hi @erikmarr, thank you for your suggestions. 
Regarding these 2 versions, they are a bit different, and namely in 2.3 each controller should implement one of the `Magento\Framework\App\Action\Http\_<Http Method>_ActionInterface` interfaces.

So it's better to leave as it is.

**Comment by erikmarr** [comment]
@eduard13 thanks for the explanation. Updates look good. I'll get this merged in today.

**Comment by erikmarr** [comment]
running tests

**Comment by erikmarr** [comment]
Thanks @eduard13!

**Comment by eduard13** [comment]
Thank you @erikmarr.

**Pull request merged**